Orange/widgets/visualize/owscatterplot.py

```diff
@@ -445,6 +446,7 @@ def _vizrank_color_change(self):
         self.vizrank_button.setToolTip(err_msg)
 
     @OWDataProjectionWidget.Inputs.data
+    @sample_dask_table
```

Comment:
That won't suffice in general. Widgets will need to the aware of both (1) the sampled data, for display, and (2) the original data, for the outputs.

Most widgets allow selection types that can be applied onto the original data. Otherwise we seriously degrade the usefulness of the outputs of these widgets that make samples.

Reply:
I agree, and I did this to enable this visualization on sampled data with as few changes as possible.

I guess we have to decide on how we approach this in general. 

I think every widget should be able to access in-memory data from Dask Table and not hold this information on its own. You would do something like this self.data.sample to access in-memory data. And I would avoid doing this separately in every widget if possible.

Reply:
I guess my question here is, do we accept the current solution as temporary until we decide on the mechanics or do we tackle this problem in this PR?